cli/internal/daemonclient/daemonclient.go

```diff
@@ -0,0 +1,53 @@
+// Package daemonclient is a wrapper around a grpc client
+// to talk to turbod
+package daemonclient
+
+import (
+	"context"
+
+	"github.com/vercel/turborepo/cli/internal/server"
+)
+
+// DaemonClient provides access to higher-level functionality from the daemon to a turbo run.
+type DaemonClient struct {
+	client server.TurboClient
+	ctx    context.Context
+}
+
+// New creates a new instance of a DaemonClient.
+func New(ctx context.Context, client server.TurboClient) *DaemonClient {
+	return &DaemonClient{
+		client: client,
+		ctx:    ctx,
+	}
+}
+
+// GetChangedOutputs implements runcache.OutputWatcher.GetChangedOutputs
+func (d *DaemonClient) GetChangedOutputs(hash string, repoRelativeOutputGlobs []string) ([]string, error) {
+	reply, err := d.client.GetChangedOutputs(d.ctx, &server.GetChangedOutputsRequest{
+		Hash:        hash,
+		OutputGlobs: repoRelativeOutputGlobs,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return reply.ChangedOutputGlobs, nil
+}
+
+// NotifyOutputsWritten implements runcache.OutputWatcher.NotifyOutputsWritten
+func (d *DaemonClient) NotifyOutputsWritten(hash string, repoRelativeOutputGlobs []string) error {
+	_, err := d.client.NotifyOutputsWritten(d.ctx, &server.NotifyOutputsWrittenRequest{
```

Comment:
I don't *think* you can define an RPC that returns `void`. Even if you can, it's not the best idea, from a compatibility perspective, since future additions to the response would cause an incompatibility with an older client. This particular RPC wouldn't matter, since we only ever call this on a same-version server, but I think it's probably better practice to just return an empty custom message rather than explain why it's ok not to, and open ourselves to the possibility of being wrong about that at some point.

See also: https://stackoverflow.com/questions/31768665/can-i-define-a-grpc-call-with-a-null-request-or-response

Reply:
Thanks for the pointer, the argument in favor of a named empty response makes sense.